src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs

```diff
@@ -46,96 +42,174 @@ public class Settings : ISettings
             new[] { "*.config" } :
             new[] { "*.Config", "*.config" };
 
-        private XDocument ConfigXDocument { get; }
-        public string FileName { get; }
-        private bool IsMachineWideSettings { get; }
-        // next config file to read if any
-        private Settings _next;
-        // The priority of this setting file
-        private int _priority;
-
         private bool Cleared { get; set; }
 
-        public Settings(string root /*, ILogger logger */)
-            : this(root, DefaultSettingsFileName, false)
+        private SettingsFile _settingsHead { get; }
+
+        private Dictionary<string, SettingsSection> _computedSections { get; set; }
+
+        public SettingsSection GetSection(string sectionName)
         {
+            if (_computedSections.TryGetValue(sectionName, out var section))
+            {
+                return section;
+            }
+
+            return null;
         }
 
-        public Settings(string root, string fileName)
-            : this(root, fileName, false)
+        public bool CreateSection(SettingsSection section, bool isBatchOperation = false)
         {
+            if (section == null)
+            {
+                throw new ArgumentNullException(nameof(section));
+            }
+
+            if (section.IsEmpty() || _computedSections.ContainsKey(section.Name))
+            {
+                return false;
+            }
+
+            var outputSettingsFile = GetOutputSettingsFileForSection(section.Name);
+
+            if (outputSettingsFile != null)
+            {
+                return outputSettingsFile.RootElement.AddChild(section, isBatchOperation);
+            }
+
+            return false;
         }
 
-        public Settings(string root, string fileName, bool isMachineWideSettings)
+        public bool SetItemInSection(string sectionName, SettingsItem item, bool isBatchOperation = false)
         {
-            if (string.IsNullOrEmpty(root))
+            if (string.IsNullOrEmpty(sectionName))
             {
-                throw new ArgumentException("root cannot be null or empty", nameof(root));
+                throw new ArgumentNullException(nameof(sectionName));
             }
 
-            if (string.IsNullOrEmpty(fileName))
+            if (item == null)
             {
-                throw new ArgumentException(Resources.Argument_Cannot_Be_Null_Or_Empty, nameof(fileName));
+                throw new ArgumentNullException(nameof(item));
             }
 
-            if (!FileSystemUtility.IsPathAFile(fileName))
+            // Check if set is an update
+            if (_computedSections.TryGetValue(sectionName, out var section))
             {
-                throw new ArgumentException(Resources.Settings_FileName_Cannot_Be_A_Path, nameof(fileName));
+                if (section.TryUpdateChildItem(item, isBatchOperation))
```

Comment:
It can be the case that the given section doesn't have that item so this will return `false`. In any case, if we were unable to update the child we then want to try to add it, not just fail the operation.